contracts/src/governance/VotingEscrowDepositor.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+interface IERC20Interface {
+    function transfer(address recipient, uint256 amount) external returns (bool);
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+    function balanceOf(address account) external view returns (uint256);
+}
+
+interface IVotingEscrowInterface {
+    function deposit_for(address _addr, uint256 _value) external;
+}
+
+contract VotingEscrowDepositor  is Ownable{
+    IERC20Interface public token;
+    IVotingEscrowInterface public escrowToken;
+    // map from address to amount availabe to withdraw
+    mapping(address => uint256) public availableToWithdraw;
+    address[] public withdrawAddresses;
+
+    constructor(address _tokenAddress, address _escrowTokenAddress){
+        token = IERC20Interface(_tokenAddress);
+        escrowToken = IVotingEscrowInterface(_escrowTokenAddress);
+    }
+
+    function increaseWithdrawAmounts(address[] memory addresses, uint256[] memory amounts) external onlyOwner{
+        require(addresses.length == amounts.length, "Arrays must be of equal length");
+        uint256 totalAmount = 0;
+        for (uint256 i = 0; i < addresses.length; i++) {
+            if (availableToWithdraw[addresses[i]] == 0) {
+                withdrawAddresses.push(addresses[i]);
+            }
+            availableToWithdraw[addresses[i]] += amounts[i];
+            totalAmount += amounts[i];
+        }
+        require(token.transferFrom(msg.sender, address(this), totalAmount), "Token transfer failed");
+    }
+
+    function clearWithdrawAmounts() external onlyOwner {
+        for (uint256 i = 0; i < withdrawAddresses.length; i++) {
+            availableToWithdraw[withdrawAddresses[i]] = 0;
+        }
+        delete withdrawAddresses;
+    }
+
+    function withdraw() external {
+        uint256 amount = availableToWithdraw[msg.sender];
+        require(amount > 0, "No amount available to withdraw");
+        availableToWithdraw[msg.sender] = 0;
+        require(token.transfer(msg.sender, amount), "Token transfer failed");
+        escrowToken.deposit_for(msg.sender, amount);
```

Comment:
This assumes someone already has an existing lock right? Do we need to check that their existing lock already has 4 years?

Reply:
it does assume that. i added a check on frontend which will direct them to create a lock if they don't have one. if they run with no lock, the transaction will fail.

this contract doesn't do any checks for the lock duration.

Reply:
Hmm, I guess it would be possible for someone to unfairly have a lower lock than someone else. Maybe we should enforce that at least their last lock date was within the last 6 months? Then there's some leeway with range, but it enforces that new locks are longer?

We'd have to tell people to use new wallets if they're contributing in new quarters? It's a little clunky.

Unless if you have a better solution? Or is there another way you had in mind of handling this situation where someone might have a 1 week lock vs 4 years?

Reply:
Another alternative is that this is handled on the "front-end" and we ask every person listed on the project to lock for X amount of time before the retroactives run? And if they don't then they just aren't added to this contract?

What do you think?

Reply:
it's pretty easy to add some changes to the frontend, which could have people extend their lock to some minimum. it wouldn't be enforced by the withdraw contract tho. i think we will move to a deposit instead of withdraw flow in the future so this could be temporary for a quarter or two.

Reply:
discussed during standup, we will have Mitchie go through setting up the lock with each contributor. we will also include the frontend check for their lock time

Reply:
added frontend check for at least 6 months in lock